src/server/services/wms/qgswmsgetcapabilities.cpp

```diff
@@ -849,18 +866,70 @@ namespace QgsWms
       layerParentElem.appendChild( treeNameElem );
     }
 
-    if ( hasQueryableChildren( projectLayerTreeRoot, QgsServerProjectUtils::wmsRestrictedLayers( *project ) ) )
+    // Instantiate CRS's from the project's crs list
+    // This will prevent us to re-instantiate all the crs's each
+    // time we will need to rebuild a bounding box.
+    auto outputCrsList = QList<QgsCoordinateReferenceSystem>();
+    for ( const QString &crsDef :  QgsServerProjectUtils::wmsOutputCrsList( *project ) )
     {
-      layerParentElem.setAttribute( QStringLiteral( "queryable" ), QStringLiteral( "1" ) );
+      auto crs = QgsCoordinateReferenceSystem::fromOgcWmsCrs( crsDef );
+      if ( crs.isValid() )
+      {
+        outputCrsList.append( crs );
+      }
     }
-    else
+
+    // Get WMS layer infos
+    const QMap< QString, LayerInfos > wmsLayerInfos = buildWmsLayerInfos( serverIface, project, outputCrsList );
+
+    const QgsRectangle wmsExtent = QgsServerProjectUtils::wmsExtent( *project );
+
+    if ( !wmsExtent.isEmpty() )
     {
-      layerParentElem.setAttribute( QStringLiteral( "queryable" ), QStringLiteral( "0" ) );
-    }
+      const QgsCoordinateReferenceSystem wgs84 = QgsCoordinateReferenceSystem::fromOgcWmsCrs( geoEpsgCrsAuthId() );
+
+      // Get WMS WGS84 bounding rectangle
+      QgsRectangle wmsWgs84BoundingRect;
+      try
+      {
+        wmsWgs84BoundingRect = transformExtent(
+                                 wmsExtent, project->crs(), wgs84, project->transformContext(), true
+                               );
+      }
+      catch ( QgsCsException &cse )
+      {
+        QgsMessageLog::logMessage( QStringLiteral( "Error transforming extent: %1" ).arg( cse.what() ), QStringLiteral( "Server" ), Qgis::MessageLevel::Warning );
+        wmsWgs84BoundingRect = QgsRectangle();
```

Comment:
I'm not sure it's necessary (the `wmsWgs84BoundingRect` still has its initial value in this situation)